terraform/modules/database/main.tf

```diff
@@ -45,7 +45,6 @@ data "aws_secretsmanager_secret" "rds_master_secret" {
 # RDS does not support secret-manager, have to provide the actual string
 data "aws_secretsmanager_secret_version" "rds_master_secret" {
   secret_id = data.aws_secretsmanager_secret.rds_master_secret.name
-  depends_on = [data.aws_secretsmanager_secret.rds_master_secret]
```

Comment:
>So you have applied that multiple times and it still keeps generating new ones?

@bmonkman yeah, i have finally been able to pin point the problem, its the depends_on
not sure I understand why though

Reply:
and i added it trying to solve the initial creation where it errors out saying 
`AWSCURRENT version not found` or something along the lines of that

Reply:
That shouldn't require `depends_on` because the line above it already creates an explicit dependency. Weird that it would cause (or fix) any problems though...